rotkehlchen/tests/api/test_history_export.py

```diff
@@ -0,0 +1,129 @@
+import csv
+import os
+from http import HTTPStatus
+from pathlib import Path
+
+
+import requests
+
+from rotkehlchen.accounting.export.csv import FILENAME_HISTORY_EVENTS_CSV
+from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.tests.utils.api import api_url_for, assert_error_response, assert_proper_response
+from rotkehlchen.tests.utils.history import prepare_rotki_for_history_processing_test
+from rotkehlchen.tests.utils.history_base_entry import add_entries
+
+
+def assert_csv_export_response(response, csv_dir, expected_count=6):
+    assert_proper_response(response)
+    data = response.json()
+    assert data['message'] == ''
+    assert data['result'] is True
+
+    # check the csv files were generated succesfully
+    with open(os.path.join(csv_dir, FILENAME_HISTORY_EVENTS_CSV), newline='', encoding='utf-8') as csvfile:  # noqa: E501
+        reader = csv.DictReader(csvfile)
+        count = 0
+        for row in reader:
+            assert len(row) == 13
+            assert row['location'] in (
+                'kraken',
+                'ethereum',
+            )
+            assert row['identifier'] is not None
+            assert row['entry_type'] is not None
+            assert row['event_identifier'] is not None
+            assert row['sequence_index'] is not None
+            assert row['timestamp'] is not None
+            assert row['asset'] is not None
+            assert row['balance_amount'] is not None
+            assert row['balance_usd_value'] is not None
+            assert row['event_type'] is not None
+            assert row['location_label'] is not None
+            assert row['notes'] is not None
+            count += 1
+        assert count == expected_count
+
+
+def test_history_export_download_csv(
+        rotkehlchen_api_server_with_exchanges,
+        tmpdir_factory,
+):
+    """Test that the csv export/download REST API endpoint works correctly."""
+    rotki = rotkehlchen_api_server_with_exchanges.rest_api.rotkehlchen
+    db = DBHistoryEvents(rotki.data.db)
+    add_entries(server=rotkehlchen_api_server_with_exchanges, events_db=db, add_directly=True)
+
+    csv_dir = str(tmpdir_factory.mktemp('test_csv_dir'))
+    csv_dir2 = str(tmpdir_factory.mktemp('test_csv_dir2'))
```

Comment:
if you use the same folder won't the file by overwritten?

Reply:
they're different folders